### PR TITLE
DATA-3397 Add GetLatestTabularData endpoint to Go SDK

### PR DESCRIPTION
**Pull request by kaywux**
App changes here: https://github.com/viamrobotics/app/pull/6896

### CODE REVIEW

**Review by stuqdog — COMMENTED** [review]
This all looks reasonable to me though we should probably resolve why tests are failing before merging.

File: go.mod [hunk 1]
```diff
@@ -76,7 +76,7 @@ require (
 	go.uber.org/atomic v1.11.0
 	go.uber.org/multierr v1.11.0
 	go.uber.org/zap v1.27.0
-	go.viam.com/api v0.1.366
+	go.viam.com/api v0.1.367
```
Comment (katiepeters) [inline go.mod:79]: This does not need to be updated. You see things failing due to TabularDataByFilter deprecation warnings

Comment (kaywux) [reply go.mod]: Ah good call - bumped back down to v0.1.366

**Review by stuqdog — APPROVED** [review]
lgtm!

**Review by katiepeters — APPROVED** [review]
Yay @kaywux! First team contribution to the Go SDK :D

File: app/data_client.go [hunk 2]
```diff
@@ -183,6 +183,14 @@ type DatabaseConnReturn struct {
 	HasDatabaseUser bool
 }
 
+// LatestTabularDataReturn represents the response returned by GetLatestTabularData. It contains the most recently captured data
+// payload, the time it was captured, and the time it was synced.
+type LatestTabularDataReturn struct {
+	TimeCaptured time.Time
+	TimeSynced   time.Time
+	Payload      *structpb.Struct
```
Comment (purplenicole730) [inline app/data_client.go:191]: We try to refrain from exposing proto types like this struct. I would make `Payload` a `[]*map[string]interface{}` type. Then define a `latestTabularDataReturnFromProto` function where you use `resp.Payload.AsMap()`.

**Review by purplenicole730 — CHANGES_REQUESTED** [review]
Made a note about avoiding giving proto types to users

File: app/data_client.go (hunk 2 above)
Comment (kaywux) [reply app/data_client.go]: Roger that, changed!

**Review by purplenicole730 — APPROVED** [review]

**Pull request merged**